homeassistant/components/sql/sensor.py

```diff
@@ -42,26 +47,24 @@ def validate_sql_select(value):
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
```

Comment:
Is the work in this function really async safe, ie no blocking I/O?

Reply:
Yes, I'd say it's required, since it's making a call to an external resource and tying up a thread while waiting for a response.

Reply:
So then we shouldn't change to async, right?

Reply:
Leaving it sync ties up a thread waiting for a callback.  That's your desired behavior?